cli/js/files.ts

```diff
@@ -200,7 +219,37 @@ export const stdout = new File(1);
 /** An instance of `File` for stderr. */
 export const stderr = new File(2);
 
-export type OpenMode =
+export interface OpenMode {
```

Comment:
A small naming suggestion here: it feels like the other concept [`CAPABILITIES(7)`](http://man7.org/linux/man-pages/man7/capabilities.7.html) at linux, and those modes should keep what the `deno` had defined. The following suggestions are to:

- rename the current `OpenMode` to `OpenCapability`
- revert the `OpenModePoxis` to `OpenMode`
- add a new polymorphism method `openSync(filename: string, cap: OpenCapability)` and the original signature `openSync(filename: string, mode: OpenMode)`

Reply:
Interesting note, but for me [CAPABILITIES(7)](http://man7.org/linux/man-pages/man7/capabilities.7.html) looks like something much broader than file opening

About TypeScript method polymorphism - it differs from strongly typed languages like C# and it's kinda weird and it has to be because of the JavaScript nature.
You have to write signatures for each of the methods and then write one implementation with signature compatible with all of the overloads. So, in this case, it would be

``` typescript
open(mode: OpenMode) : File;
open(mode: OpenModePosix): File;
open(mode: X) : File {
// implementation with mode type check
}
````
where X have to be compatibile with both modes, so It would be either `any` or `OpenMode | OpenModePosix` so basically same thing as already implemented